### PR TITLE
`TextEdit`: Change `margin` property to `egui::Margin` type

### PR DESCRIPTION
**Pull request by bu5hm4nn**
Despite their being an actual `egui::Margin` struct, Textedit has a `margin()` builder function that supports only `Vec2` types and thereby only symmetric margins. This PR changes the function to accept `egui::Margin` type instead making it more congruent with overall egui logic as well as supporting asymmetric margins.

P.S: I tried to run all checks but I had to modify `./rust-toolchain` to 1.67.0 to get the checks to run on macOS.

### CODE REVIEW

File: crates/egui/src/widgets/text_edit/builder.rs [hunk 1]
```diff
@@ -239,7 +239,7 @@ impl<'t> TextEdit<'t> {
     }
 
     /// Set margin of text. Default is [4.0,2.0]
-    pub fn margin(mut self, margin: Vec2) -> Self {
+    pub fn margin(mut self, margin: Margin) -> Self {
         self.margin = margin;
```
Comment (emilk) [inline crates/egui/src/widgets/text_edit/builder.rs:243]:
```suggestion
    #[inline]
    pub fn margin(mut self, margin: impl Into<Margin>) -> Self {
        self.margin = margin.into();
```

File: crates/egui/src/widgets/text_edit/builder.rs [hunk 2]
```diff
@@ -340,11 +340,18 @@ impl<'t> TextEdit<'t> {
         let where_to_put_background = ui.painter().add(Shape::Noop);
 
         let margin = self.margin;
-        let max_rect = ui.available_rect_before_wrap().shrink2(margin);
+        let available = ui.available_rect_before_wrap();
+        let max_rect = Rect::from_x_y_ranges(
+            available.left() + margin.left..=available.right() - margin.right,
+            available.top() + margin.top..=available.bottom() - margin.bottom,
+        );
```
Comment (emilk) [inline crates/egui/src/widgets/text_edit/builder.rs:347]:
```suggestion
        let max_rect = margin.shrink_rect(available);
```

File: crates/egui/src/widgets/text_edit/builder.rs [hunk 3]
```diff
@@ -340,11 +340,18 @@ impl<'t> TextEdit<'t> {
...
         let mut content_ui = ui.child_ui(max_rect, *ui.layout());
         let mut output = self.show_content(&mut content_ui);
         let id = output.response.id;
-        let frame_rect = output.response.rect.expand2(margin);
+        let frame_rect = Rect::from_x_y_ranges(
+            output.response.rect.left() - margin.left..=output.response.rect.right() + margin.right,
+            output.response.rect.top() - margin.top..=output.response.rect.bottom() + margin.bottom,
+        );
```
Comment (emilk) [inline crates/egui/src/widgets/text_edit/builder.rs:354]:
```suggestion
        let frame_rect = margin.expand_rect(output.response.rect);
```

File: crates/egui/src/widgets/text_edit/builder.rs [hunk 4]
```diff
@@ -430,7 +437,7 @@ impl<'t> TextEdit<'t> {
             available_width
         } else {
             desired_width.min(available_width)
-        } - margin.x * 2.0;
+        } - (margin.left + margin.right);
```
Comment (emilk) [inline crates/egui/src/widgets/text_edit/builder.rs:440]:
```suggestion
        } - margin.sum().x;
```

File: crates/egui/src/widgets/text_edit/builder.rs [hunk 5]
```diff
@@ -453,8 +460,12 @@ impl<'t> TextEdit<'t> {
             galley.size().x.max(wrap_width)
         };
         let desired_height = (desired_height_rows.at_least(1) as f32) * row_height;
-        let desired_size = vec2(desired_width, galley.size().y.max(desired_height))
-            .at_least(min_size - margin * 2.0);
+        let at_least = Vec2::new(
+            min_size.x - (margin.left + margin.right),
+            min_size.y - (margin.top + margin.bottom),
+        );
```
Comment (emilk) [inline crates/egui/src/widgets/text_edit/builder.rs:466]:
```suggestion
        let at_least = min_size - margin.sum();
```

**Review by emilk — CHANGES_REQUESTED** [review]
Thanks!

File: crates/egui/src/widgets/text_edit/builder.rs (hunk 1 above)
Comment (bu5hm4nn) [reply crates/egui/src/widgets/text_edit/builder.rs]: Resolved by merge of latest master

**Review by emilk — APPROVED** [review]

**Comment by emilk** [comment]
The code doesn't compile yet :)

File: crates/egui/src/widgets/drag_value.rs [hunk 6]
```diff
@@ -490,7 +490,10 @@ impl<'a> Widget for DragValue<'a> {
                     .clip_text(false)
                     .horizontal_align(ui.layout().horizontal_align())
                     .vertical_align(ui.layout().vertical_align())
-                    .margin(ui.spacing().button_padding)
+                    .margin(Margin::symmetric(
+                        ui.spacing().button_padding.x,
+                        ui.spacing().button_padding.y,
+                    ))
```
Comment (emilk) [inline crates/egui/src/widgets/drag_value.rs:496]: `Margin should implement `From<Vec2>` so that the old code still compiles

**Review by emilk — APPROVED** [review]

File: crates/egui/src/widgets/drag_value.rs (hunk 6 above)
Comment (bu5hm4nn) [reply crates/egui/src/widgets/drag_value.rs:496]: Sure, I can add that.

Comment (bu5hm4nn) [reply crates/egui/src/widgets/drag_value.rs:496]: Already exists: https://github.com/emilk/egui/blob/d22ee81ab02a96ff2cfa15903b0d4255bec55962/crates/egui/src/style.rs#L684

Comment (emilk) [inline crates/egui/src/widgets/drag_value.rs:496]:
```suggestion
                    .margin(ui.spacing().button_padding)
```

**Review by emilk — APPROVED** [review]

**Pull request merged**